op-e2e/actions/user_test.go

```diff
@@ -12,17 +13,42 @@ import (
 	"github.com/ethereum-optimism/optimism/op-node/testlog"
 )
 
-// TestCrossLayerUser tests that common actions of the CrossLayerUser actor work:
+type regolithScheduledTest struct {
+	name             string
+	regolithTime     *hexutil.Uint64
+	activateRegolith bool
+}
+
+// TestCrossLayerUser tests that common actions of the CrossLayerUser actor work in various regolith configurations:
 // - transact on L1
 // - transact on L2
 // - deposit on L1
 // - withdraw from L2
 // - prove tx on L1
 // - wait 1 week + 1 second
 // - finalize withdrawal on L1
-func TestCrossLayerUser(gt *testing.T) {
+func TestCrossLayerUser(t *testing.T) {
+	zeroTime := hexutil.Uint64(0)
+	futureTime := hexutil.Uint64(20)
+	farFutureTime := hexutil.Uint64(2000)
+	tests := []regolithScheduledTest{
+		{name: "NoRegolith", regolithTime: nil, activateRegolith: false},
+		{name: "NotYetRegolith", regolithTime: &farFutureTime, activateRegolith: false},
+		{name: "RegolithAtGenesis", regolithTime: &zeroTime, activateRegolith: true},
+		{name: "RegolithAfterGenesis", regolithTime: &futureTime, activateRegolith: true},
+	}
+	for _, test := range tests {
+		testCfg := test // Use a fixed reference as the tests run in parallel
```

Comment:
They don't run in parallel yet, but you could call `t.Parallel()` below inside `t.Run` to make them.

Reply:
Turns out they do run in parallel - the call is hidden down in some of the setup calls. It took me a while to work out...